skytemple_randomizer/randomizer/npc.py

```diff
@@ -91,6 +107,160 @@ def run(self, status: Status):
 
         status.done()
 
+    def _smart_replace_text(self, mapped_actor_names_by_lang):
+        for lang, lang_string_file in get_all_string_files(self.rom, self.static_data):
+            mapped_actor_names = mapped_actor_names_by_lang[lang]
+            sorted_actor_names = sorted(list(mapped_actor_names.keys()), key=len, reverse=True)
+            # Most NPC texts in the base game are wrapped via [CN:N]...[CR], or [CN:Y]...[CR].
+            # Some place names derived from NPCs are mentioned via [CS:P]...[CR], so we'll replace those as well.
+            # Croagunk's Swap Shop is mentioned via [CS:E].
+            standard_npc_text = re.compile(
```

Comment:
Previously this was two different regexes for (N|Y|E) and (P), however I've noticed that some European language versions have E-colour strings that include more than just the pokemon, e.g.:

English: `Finding an [CS:Z]Ariados Bow[CR] at the [CS:E]Croagunk[CR] Swap Shop is very lucky!`
French: `Dénicher un [CS:Z]Migaloruban[CR] au [CS:E]Stand Troc Cradopaud[CR] est un sacré coup de chance!`

Rather than maintaining multiple regexes I've opted to consolidate them into one - note that sorting the actor names is important as it will ensure the longest names are matched first, e.g.:

![image](https://github.com/SkyTemple/skytemple-randomizer/assets/15063274/abc9ca15-1874-4e2b-8d96-9f31286b5aa3)

The regex has also been updated to use `((?:\s|.)*?)` for the prefix and suffix parts of the replacement (may be empty), as some of these span across newlines.